E01-Geometry/Dovydas/src/Kilo/Square.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Kilo;
+class Square implements Shape
+{
+    private float $length;
+    private float $height;
+    private Geometry $geometry;
+
+    /**
+     * Square constructor.
+     * @param float $length
+     * @param float $height
+     */
+    public function __construct(float $length, float $height)
+    {
+        $this->length = $length;
+        $this->height = $height;
+        $this->setGeometry();
+    }
+
+
+    public function setGeometry(): void
+    {
+        $this->geometry = new SquareGeometry($this);
```

Comment:
Almost never `new` your dependencies in your class.
Use `dependency injection`.


But in this case, I think whole `XGeometry` is a bit unnecessary. 
Just let the `Shape` calculates their area and perimeter.